apps/rush-lib/src/cli/actions/VersionAction.ts

```diff
@@ -94,6 +94,7 @@ export class VersionAction extends BaseRushAction {
   }
 
   protected run(): Promise<void> {
+    VersionMismatchFinder.runRushCheckIfNecessary(this.rushConfiguration);
```

Comment:
>VersionMismatchFinder.runRushCheckIfNecessary(this.rushConfiguration); [](start = 4, length = 70)

I'm a little concerned about a scenario where we:

1. We publish Rush with this feature.
2. We enable this in our master branch, and start doing builds with it.
3. Later, the daily job CI comes along `rush version` fails for some reason due to a weird thing that only happens in release branches.
4. Because there's no `--bypass-policy` we have no quick way to mitigate this.  We need to revert the option in various branches, and file a bug for `rush version`

Options:
- Ask @qz2017 for some confidence that this won't happen
- Provide a way to bypass it
- Don't enforce this for `rush version` and `rush publish`

Reply:
I'll go with option 3.

---
In reply to: [218996862](https://github.com/Microsoft/web-build-tools/pull/829#discussion_r218996862) [](ancestors = 218996862)